src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -0,0 +1,111 @@
+package ui.components.pickers;
+
+import util.Utility;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class MilestonePickerState {
+    private List<PickerMilestone> currentMilestonesList;
+
+    public MilestonePickerState(List<PickerMilestone> milestones) {
+        currentMilestonesList = cloneList(milestones);
+    }
+
+    public MilestonePickerState(List<PickerMilestone> milestones, String userInput) {
+        this(milestones);
+        processInput(userInput);
+    }
+
+    private List<PickerMilestone> cloneList(List<PickerMilestone> sourceList) {
+        List<PickerMilestone> destList = new ArrayList<>();
+        sourceList.stream()
+                .forEach(milestone -> destList.add(new PickerMilestone(milestone)));
+        return destList;
```

Comment:
I don't think so, that will make it refer to the same object right? With each change in input field, the original list of milestones is re-supplied, so we can't mutate them.